spec/controllers/groups_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails.  There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+#
+# Compared to earlier versions of this generator, there is very limited use of
+# stubs and message expectations in this spec.  Stubs are only used when there
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+
+RSpec.describe GroupsController, type: :controller do
+
+  # This should return the minimal set of attributes required to create a valid
+  # Group. As you add validations to Group, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+    skip("Add a hash of attributes valid for your model")
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # GroupsController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "assigns all groups as @groups" do
+      group = Group.create! valid_attributes
+      get :index, {}, valid_session
+      expect(assigns(:groups)).to eq([group])
+    end
+  end
+
+  describe "GET #show" do
+    it "assigns the requested group as @group" do
+      group = Group.create! valid_attributes
+      get :show, {:id => group.to_param}, valid_session
+      expect(assigns(:group)).to eq(group)
+    end
+  end
+
+  describe "GET #new" do
+    it "assigns a new group as @group" do
+      get :new, {}, valid_session
+      expect(assigns(:group)).to be_a_new(Group)
+    end
+  end
+
+  describe "GET #edit" do
+    it "assigns the requested group as @group" do
+      group = Group.create! valid_attributes
+      get :edit, {:id => group.to_param}, valid_session
+      expect(assigns(:group)).to eq(group)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new Group" do
+        expect {
+          post :create, {:group => valid_attributes}, valid_session
+        }.to change(Group, :count).by(1)
+      end
+
+      it "assigns a newly created group as @group" do
+        post :create, {:group => valid_attributes}, valid_session
+        expect(assigns(:group)).to be_a(Group)
+        expect(assigns(:group)).to be_persisted
+      end
+
+      it "redirects to the created group" do
+        post :create, {:group => valid_attributes}, valid_session
+        expect(response).to redirect_to(Group.last)
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns a newly created but unsaved group as @group" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.